trunner/testcase.py

```diff
@@ -176,15 +177,17 @@ def create(test):
             return TestCaseUnit(
                     name=test['name'],
                     target=test['target'],
-                    exec_bin=test['syspage'],
+                    timeout=test['timeout'],
+                    exec_bin=test['exec'],
                     status=status
             )
         if test['type'] == 'harness':
             return TestCaseCustomHarness(
                     name=test['name'],
                     target=test['target'],
+                    timeout=test['timeout'],
```

Comment:
timeout is a good idea, but I would expect it's a timeout for the whole test, not for each expect() separately. I can't think of a better name straight away.

Btw, I don't know if we need full-test timeout for now, but maybe in the future we will. We may also add total test execution time in the test status output (right now in CI I don't know what took the most time).